### ISSUE TITLE
tc-tiddler-body a.tc-tiddlylink is applied too often

### ISSUE THREAD

**Issue by CodaCodr** [TheDiveO/TW5FontAwesome#3]
I'm seeing an issue with the current release which shows up as 1.1.0 on my system.

The "cog" or "gear" applied to system tiddlers can show up repeatedly next to (before) a link (three times in the links listed in my $:/Import tiddler).

I modified $:/plugins/TheDiveO/FontAwesome/styles/tiddlylinks/system as a workaround.

Clue: perhaps this css doesn't play nice with PMario's unilink plugin? (just a guess)

**Comment by thediveo** [comment]
Do you have a test case to reproduce? Without one it's difficult for me to reproduce and fix. What happens when you temporarily disable the unilink plugin, is the gear/cog visible only once?

**Comment by CodaCodr** [comment]
Yes, sorry, I did try that and the issue disappears when unilink is disabled.

**Comment by thediveo** [comment]
I've looked at the HTML markup/DOM elements that unilink creates and this appears to me to be broken: for every "unilink" the plugin creates first an empty(!) `a` element, then followed by the correctly populated `a` element. Both elements have the correct CSS classes, but the first empty `a` element doesn't make sense to me as you neither can click it using the mouse nor hit it with a finger touch. It's empty, so it doesn't help, but triggers the CSS stylesheet rule for the link adornment.

For your reference, this is what the unilink plugin renders when I view it in Firefox:

<img width="697" alt="broken unilink markup" src="https://user-images.githubusercontent.com/6920158/34887953-a5ed4c3a-f7c8-11e7-95c6-dd0b9210d7e1.png">

So my FontAwesome CSS styling correctly gets applied twice, because of the odd (or unoptimized?) `a` element markup that uni-link produces. This is a problem you should report to the author of the uni-link plugin.

**Comment by CodaCodr** [comment]
Agreed.  Thanks for your diligence!

**Comment by thediveo** [comment]
You're welcome. I thought about some CSS hack around using `:not` and `:empty` but other CSS will trigger this again. I suspect that the hack in @wikilabs unilink to get automatic backlinking may cause the unwanted side-effect on CSS `:before` rules, because it emits an "invisible" empty link.

**Comment by CodaCodr** [comment]
It probably also explains why *sometimes* I bump into another issue using the $listWidget where, no matter what I do, I end up with a block(div) when I'm specifying inline(span) or the reverse.  Drives me crazy.  It doesn't happen often thankfully, but when I see it happening and check in the inspector, excessive a tags - nested, too.

I'll drop a note for Mario.  Thanks again.

**Comment by pmario** [comment]
Hi @CodaCodr 
If you would have mentioned me with @pmario, I would have seen the issue. 

It seems I don't get a notification with @wikilabs, ... That's strange.

**Comment by pmario** [comment]
@CodaCodr ... can you **please test** the latest plugin v0.3.0? https://wikilabs.github.io/editions/uni-link/

It's a temporary fix. The code still produces a redundant element `x` which should be ignored by all browsers. ... But TW 5.1.15 still applys some classes. 5.1.16 has a much better behaviour, due to a different PR that has been merged. ... 

I did create a new issue. https://github.com/wikilabs/plugins/issues/9
